src/Microsoft.AspNetCore.Razor.Evolution/TagHelperBinderSyntaxTreePass.cs

```diff
@@ -41,10 +45,13 @@ public RazorSyntaxTree Execute(RazorCodeDocument codeDocument, RazorSyntaxTree s
             
             visitor.VisitBlock(syntaxTree.Root);
 
-            var directives = visitor.Directives;
             var errorSink = new ErrorSink();
-            var descriptors = resolver.Resolve(new TagHelperDescriptorResolutionContext(directives, errorSink)).ToArray();
-            if (descriptors.Length == 0)
+            var directives = visitor.Directives;
+            var descriptors = (IReadOnlyList<TagHelperDescriptor>)resolver.Resolve(errorSink).ToList();
```

Comment:
Yeah I could go either way... opting to leave it because it's the smaller change.